variables.tf

```diff
@@ -205,3 +190,87 @@ variable "permissions_boundary" {
 }
+
+##################
+# All the following variables are just about configuring the Kubernetes provider
+# to be able to modify the aws-auth ConfigMap. Once EKS provides a normal
+# AWS API for modifying it, we can do away with all of this.
+#
+# The reason there are so many options is because at various times, each
+# one of them has had problems, so we give you a choice.
+#
+# The reason there are so many "enabled" inputs rather than automatically
+# detecting whether or not they are enabled based on the value of the input
+# is that logic based in input values requires the values be known during
+# the "plan" phase of Terraform, which often causes problems.
+#
+
+variable "kubeconfig_path_enabled" {
+  type        = bool
+  default     = false
+  description = "If `true`, configure the Kubernetes provider with `kubeconfig_path` and use it for authenticating to the EKS cluster"
+}
+
+variable "kubeconfig_path" {
+  type        = string
+  default     = ""
+  description = "The Kubernetes provider `config_path` setting to use when `kubeconfig_path_enabled` is `true`"
+}
+
+variable "kube_data_auth_enabled" {
+  type        = bool
+  default     = true
+  description = <<-EOT
+    If `true`, use an `aws_eks_cluster_auth` data source to authenticate to the EKS cluster.
+    Disabled by `kubeconfig_path_enabled`, overrides `kube_exec_auth_enabled`.
+    EOT
+}
+
+variable "kube_exec_auth_enabled" {
+  type        = bool
+  default     = false
+  description = <<-EOT
+    If `true`, execute `aws eks get-token` to authenticate to the EKS cluster.
+    Disabled by `kubeconfig_path_enabled` or `kube_exec_auth_enabled`.
+    EOT
+}
+
+
+variable "kube_exec_auth_role_arn" {
+  type        = string
+  default     = ""
+  description = "The role ARN for `aws eks get-token` to use"
+}
+
+variable "kube_exec_auth_role_arn_enabled" {
+  type        = bool
+  default     = false
+  description = "If true, pass `kube_exec_auth_role_arn` as the role ARN to `aws eks get-token`"
+}
+
+variable "kube_exec_auth_aws_profile" {
+  type        = string
+  default     = ""
+  description = "The AWS config profile for `aws eks get-token` to use"
+}
+
+variable "kube_exec_auth_aws_profile_enabled" {
+  type        = bool
+  default     = false
+  description = "If true, pass `kube_exec_auth_aws_profile` as the `profile` to `aws eks get-token`"
+}
+
+variable "aws_auth_yaml_strip_quotes" {
+  type        = bool
+  default     = true
+  description = "If true, remove double quotes from the generated aws-auth ConfigMap YAML to reduce suprious diffs in plans"
+}
+
+variable "dummy_kubeapi_server" {
+  type        = string
+  default     = "https://jsonplaceholder.typicode.com"
+  description = <<-EOT
+    URL of a dummy API server for the Kubernetes server to use when the real one is unknown.
+    This is a workaround to ignore connection failures when the results do not matter.
```

Comment:
```suggestion
    URL of a dummy API server for the Kubernetes server to use when the real one is unknown.
    This is a workaround to ignore connection failures when the results do not matter.
    You can disable it by setting it to `null`; however, doing so _will_ cause Kubernetes upgrades run by terraform to fail.
```

Reply:
I updated the comment with more information